pkg/windows/windows.go

```diff
@@ -335,6 +344,203 @@ func (vm *windows) Configure() error {
+			return ipAddresses, nil
+		}
+		// unknown error, return
+		return nil, err
+	}
+	// trim leading whitespaces and trailing CR LF (\r\n)
+	out = strings.TrimSpace(out)
+	// parse, where the separator is the new line escape sequence
+	// in Windows. See https://en.wikipedia.org/wiki/Newline#Representation
+	tokens := strings.Split(out, "\r\n")
+	// loop, cleanup and collect
+	for _, ip := range tokens {
+		// trim
+		ip = strings.TrimSpace(ip)
+		// check empty values
+		if ip == "" {
+			// skip
+			continue
+		}
+		// collect
+		ipAddresses = append(ipAddresses, ip)
+	}
+	// return
+	return ipAddresses, nil
+}
+
+// getApiServerInternalIpForPlatformStatus returns the APIServerInternalIP given the platform type
+//
+// Currently, VSphere is the only supported platform with the explicit APIServerInternalIP field,
+// as WMCO becomes available to other platforms (BareMetal, OpenStack, Ovirt, Kubevirt) support should be enabled
+// https://docs.openshift.com/container-platform/4.7/rest_api/config_apis/infrastructure-config-openshift-io-v1.html
+func getApiServerInternalIpForPlatformStatus(platformStatus *config.PlatformStatus) (string, error) {
+	if platformStatus == nil {
+		return "", errors.New("platformStatus cannot be nil")
+	}
+	switch platformStatus.Type {
+	case config.VSpherePlatformType:
+		return platformStatus.VSphere.APIServerInternalIP, nil
+		// TODO: uncomment as platforms become supported
+		// case config.BareMetalPlatformType:
+		// 	return platformStatus.BareMetal.APIServerInternalIP, nil
+		// case config.OpenStackPlatformType:
+		// 	return platformStatus.OpenStack.APIServerInternalIP, nil
+		// case config.OvirtPlatformType:
+		// 	return platformStatus.Ovirt.APIServerInternalIP, nil
+		// case config.KubevirtPlatformType:
+		// 	return platformStatus.Kubevirt.APIServerInternalIP, nil
+	}
+	// error, not supported
+	return "", errors.Errorf("cannot find APIServerInternalIP for platform type %s",
+		platformStatus.Type)
+}
+
+// ensureAPIServerInternal ensures the resolution of the APIServerInternalIP
+//
+// Usually, the APIServerInternalIP relates to the hostname with prefix "api-int."
+func (vm *windows) ensureAPIServerInternal() error {
+	// get infra status
+	status, err := getClusterInfrastructureStatus()
+	if err != nil {
+		return err
+	}
+	// get internal URL
+	apiServerInternalURL := status.APIServerInternalURL
+	// parse hostname
+	apiServerInternalHostname, err := parseHostname(apiServerInternalURL)
+	if err != nil {
+		return err
+	}
+	// flush DNS cache
+	if err := vm.clearDnsClientCache(); err != nil {
+		return err
+	}
+	vm.log.Info("Finding API server internal IP address for hostname",
+		"hostname", apiServerInternalHostname,
+		"platformType", status.PlatformStatus.Type)
+	// check existing DNS resolution for apiServerInternalHostname including
+	// existing hosts entries that may be pre-configured in the golden image
+	ipAddresses, err := vm.resolveIpAddresses(apiServerInternalHostname, false)
```

Comment:
Does it makes sense to ignore any existing hosts file entries, that may be pre-configured in the golden image?